CPPLogger/CPPLogger.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <iostream>
+#include <string>
+
+#include "P7_Client.h"
+#include "P7_Trace.h"
+
+
+enum LogLevels
+{
+	L_TRACE = 0,
+	L_DEBUG,
+	L_INFO,
+	L_WARNING,
+	L_ERROR,
+	L_CRITICAL,
+};
+
+class ILogger
+{
+public:
+	virtual ILogger& operator() (const std::string& func_name, const std::string& file_name, short level) = 0;
+	virtual ILogger& operator<< (const char* log_message) = 0;
+	virtual void set_filter_level(LogLevels level) = 0;
+};
+
+class Logger : public ILogger
```

Comment:
If you will not inherit from the class "Logger" put keyword "final".
Like this:
class Logger final : public ILogger